src/SAWScript/Crucible/JVM/Builtins.hs

```diff
@@ -268,20 +270,24 @@ verifyObligations cc mspec tactic assumes asserts =
      let nm = mspec ^. csMethodName
      stats <- forM (zip [(0::Int)..] asserts) $ \(n, (msg, assert)) -> do
        goal   <- io $ scImplies sc assume assert
-       goal'  <- io $ predicateToProp sc Universal goal
+       goal'  <- io $ boolToProp sc [] goal -- TODO, generalize over inputs
        let goalname = concat [nm, " (", takeWhile (/= '\n') msg, ")"]
            proofgoal = ProofGoal n "vc" goalname goal'
-       r      <- evalStateT tactic (startProof proofgoal)
-       case r of
-         Unsat stats -> return stats
-         SatMulti stats vals -> do
+       res <- runProofScript tactic proofgoal
+       case res of
+         ValidProof stats _thm -> return stats -- TODO, do something with these theorems!
```

Comment:
It might be valuable to store theorems in `MethodSpec`s? We'd want to check that it doesn't lead to significantly increased memory use, though, and it doesn't have to happen as part of this PR.

Reply:
Agreed.  I discuss this a bit in https://github.com/GaloisInc/saw-script/issues/1136